core/src/main/java/io/parsingdata/metal/data/selection/ByPredicate.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2013-2016 Netherlands Forensic Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.parsingdata.metal.data.selection;
+
+import static io.parsingdata.metal.SafeTrampoline.complete;
+import static io.parsingdata.metal.SafeTrampoline.intermediate;
+
+import java.util.function.Predicate;
+
+import io.parsingdata.metal.SafeTrampoline;
+import io.parsingdata.metal.data.ImmutableList;
+import io.parsingdata.metal.data.ParseGraph;
+import io.parsingdata.metal.data.ParseItem;
+import io.parsingdata.metal.data.ParseValue;
+
+public final class ByPredicate {
+
+    private ByPredicate() {}
+
+    public static final int NO_LIMIT = -1;
+
+    static SafeTrampoline<ImmutableList<ParseValue>> getAllValues(final ImmutableList<ParseGraph> graphList, final ImmutableList<ParseValue> valueList, final Predicate<ParseValue> predicate, final int limit) {
+        if (graphList.isEmpty() || valueList.size == limit) { return complete(() -> valueList); }
```

Comment:
I think we have tests for "get all" and "get last", but no explicit test for something like "get the last 3". Maybe add that?

Also: this is not (yet?) exposed to the `ref()` and `token()` expressions, right? Is there an issue yet to add that?